packages/material-ui/src/ClickAwayListener/ClickAwayListener.tsx

```diff
@@ -63,7 +109,11 @@ function ClickAwayListener(props) {
     // 1. IE11 support, which trigger the handleClickAway even after the unbind
     // 2. The child might render null.
     // 3. Behave like a blur listener.
-    if (!activatedRef.current || !nodeRef.current || clickedRootScrollbar(event, doc)) {
+    if (
+      !activatedRef.current ||
+      !nodeRef.current ||
+      ('clientX' in event && clickedRootScrollbar(event, doc))
```

Comment:
We used to access `event.clientX` on touch events which don't implement that property. So we would always end up evaluating `doc.documentElement.clientWidth < undefined || doc.documentElement.clientHeight < undefined`. This always evaluates to `false` so we can bail out early by checking if we have a MouseEvent.

In the end we're forcing less layout on touch events (no access to `element.clientWidth`) but check property existence more often. Seems like a good tradeoff considering forcing layout is much more expensive than checking property existence and touch events are usually dispatched on weaker devices e.g. smartphones.